src/Heightmap.cc

```diff
@@ -311,7 +353,23 @@ Errors Heightmap::Load(ElementPtr _sdf)
 
   if (_sdf->HasElement("uri"))
   {
-    this->dataPtr->uri = _sdf->Get<std::string>("uri", "").first;
+    auto uri = _sdf->Get<std::string>("uri", "").first;
+
+    if (_config.StoreResolvedURIs())
+    {
+      const std::string resolvedUri = sdf::findFile(uri, true, true, _config);
+      if (resolvedUri.empty())
+      {
+        errors.push_back({ErrorCode::URI_LOOKUP,
+            "Parser configurations requested resolved uris, but uri ["
+            + uri + "] could not be resolved."});
+      }
+      else
+      {
+        uri = resolvedUri;
+      }
+    }
```

Comment:
consider a helper function for this code block that is repeated 3 times in this file

Reply:
Added a helper to utils, and used it accordingly.